lib/svrquery/protocol/sqp/query.go

```diff
@@ -420,58 +422,43 @@ func (q *queryer) readQueryTeamInfo(qr *QueryResponse, r *packetReader) (err err
 	return nil
 }
 
-func (q *queryer) readQueryMultiPacket(version uint16, curPkt, lastPkt, requestedChunks byte, pktLen uint16) (*QueryResponse, error) {
-	// Setup our array of packet bodies
-	expectedPkts := lastPkt + 1
-	multiPkt := make(map[byte][]byte, expectedPkts)
-	totalPktLen := uint32(pktLen)
+func (q *queryer) readQueryMetrics(qr *QueryResponse, r *packetReader) (err error) {
+	qr.Metrics = &MetricsChunk{}
 
-	// Handle this first packet
-	multiPkt[curPkt] = make([]byte, pktLen)
-	n, err := q.reader.Read(multiPkt[curPkt])
-	if err != nil {
-		return nil, err
-	} else if uint16(n) != pktLen {
-		return nil, NewErrMalformedPacketf("expected packet length of %v, but read %v bytes", pktLen, n)
+	if qr.Metrics.ChunkLength, err = r.ReadUint32(); err != nil {
+		return err
 	}
+	l := int64(qr.Metrics.ChunkLength)
 
-	// Remember the challengeID so that we can verify each packet we are reading is
-	// part of this multi-packet response
-	challengeID := q.challengeID
-
-	// Handle each subsequent packet until we have all of the ones we need
-	for len(multiPkt) != int(expectedPkts) {
-		version, curPkt, lastPkt, pktLen, err = q.readQueryHeader()
-		if err != nil {
-			return nil, err
-		}
+	qr.Metrics.MetricCount, err = r.ReadByte()
+	if err != nil {
+		return err
+	}
+	l -= int64(Byte.Size())
 
-		// If this packet isn't part of the multi-packet response we are expecting, discard it
-		if q.challengeID != challengeID {
-			if _, err := io.CopyN(ioutil.Discard, q.reader, int64(pktLen)); err != nil {
-				return nil, err
-			}
-			continue
-		}
+	if qr.Metrics.MetricCount > MaxMetrics {
+		qr.Metrics.MetricCount = MaxMetrics
```

Comment:
Wouldn't changing this cause the read from the below metrics read potentially behave strangely? For example if this was not the last chunk type. Would it be better erroring?

Reply:
Yeah, makes sense. Fixed.